Library/UIKit/UIViewController+StoryboardSegueIdentifierProtocol.swift

```diff
@@ -9,16 +9,29 @@
 import Foundation
 import UIKit
 
-public extension UIViewController {
+public protocol UIViewControllerType {
```

Comment:
No, it can't be an extension on UIViewController, because of the `Self` requirement:

> `Self` is only available in a protocol or as the result of a method in a class

So that's why a protocol is needed. I'm unsure if there is some fundamental type system reason for this. Or simply a limitation of the current Swift compiler implementation. If its a "bug" in the compiler, we can assume it will be fixed at some point in the future, and then we can remove this protocol.

And yeah, I don't really like the name either. I thought of it because of things like `CollectionType` and `SequenceType`.
¯\_(ツ)_/¯